channeldb/channel.go

```diff
@@ -2068,11 +2072,25 @@ type HTLC struct {
 	// from the HtlcIndex as this will be incremented for each new log
 	// update added.
 	LogIndex uint64
+
+	// ExtraData contains any additional information that was transmitted
+	// with the HTLC via TLVs. This data *must* already be encoded as a
+	// TLV stream, and may be empty.
+	ExtraData []byte
 }
 
 // SerializeHtlcs writes out the passed set of HTLC's into the passed writer
 // using the current default on-disk serialization format.
 //
+// This inline serialization has been extended to allow storage of extra data
+// associated with a HTLC in the following way:
+//   - The known-length onion blob (1366 bytes) is serialized as var bytes in
+//     WriteElements (ie, the length 1366 was written, followed by the 1366
+//     onion bytes).
+//   - To include extra data, we append any extra data present to this one
```

Comment:
Nit: Maybe add a comment that the Size Limit of the Extra_Data is limited by the protocol max message size minus the onion blob?

Reply:
That's not strictly true, it's limited to the space available to TLVs in `update_add_htlc` which needs space for its other fields. Will add a comment.